### ISSUE TITLE
Inconsistent Counts using "cassandra" or "astyanax"

### ISSUE THREAD

**Issue by thomascramer** [thinkaurelius/titan#949]
I'm getting inconsistent counts when connecting titan to a cassandra cluster using the "storage.backend" to "cassandra" or "astyanax" though "cassandrathrift" seems to work just fine (I've created a mailing list topic to this effect https://groups.google.com/forum/#!topic/aureliusgraphs/2pnIGr32XTg).  It seems more prevalent when starting to do a 5,000 or 6,000 data set.  

When connecting via gremlin or rexster with "cassandra" or "astyanax" as the storage driver I get:

rexster[groovy]> g.V.count()
==>5888
rexster[groovy]> g.V.count()
==>5869
rexster[groovy]> g.V.count()
==>5838
rexster[groovy]> g.V.count()
==>5879

**Comment by mbroecheler** [comment]
Yes, we have also observed some strange "caching" behavior for Astyanax. We
don't have an explanation for that yet.

On Thu, Jan 29, 2015 at 10:11 AM, Thomas Cramer notifications@github.com
wrote:

> I'm getting inconsistent counts when connecting titan to a cassandra
> cluster using the "storage.backend" to "cassandra" or "astyanax" though
> "cassandrathrift" seems to work just fine (I've created a mailing list
> topic to this effect
> https://groups.google.com/forum/#!topic/aureliusgraphs/2pnIGr32XTg). It
> seems more prevalent when starting to do a 5,000 or 6,000 data set.
> 
> When connecting via gremlin or rexster with "cassandra" or "astyanax" as
> the storage driver I get:
> 
> rexster[groovy]> g.V.count()
> ==>5888
> rexster[groovy]> g.V.count()
> ==>5869
> rexster[groovy]> g.V.count()
> ==>5838
> rexster[groovy]> g.V.count()
> ==>5879
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/thinkaurelius/titan/issues/949.

## 

Matthias Broecheler
http://www.matthiasb.com

**Comment by frensjan** [comment]
Might this be related to: [https://issues.apache.org/jira/plugins/servlet/mobile#issue/CASSANDRA-8940]?

**Comment by ghost** [comment]
Do we know if the Cassandra defect referenced addressed this problem or if there is still a concern about astyanax caching or some other unkown cause ?
